src/utils.rs

```diff
@@ -93,55 +86,86 @@ mod unsafe_code {
 #[cfg(feature = "allow-unsafe-code")]
 pub use unsafe_code::CSlice;
 
-/// A simple wrapper around RawFd that closes the fd on drop.
-#[derive(Debug, Hash, PartialEq, Eq)]
-pub struct RawFdContainer(RawFd);
+#[cfg(unix)]
+mod raw_fd_container {
+    use std::mem::forget;
+    #[cfg(unix)]
+    use std::os::unix::io::{AsRawFd, IntoRawFd, RawFd};
+
+    /// A simple wrapper around RawFd that closes the fd on drop.
+    ///
+    /// On non-unix systems, this type is empty and does not provide
+    /// any method.
+    #[derive(Debug, Hash, PartialEq, Eq)]
+    pub struct RawFdContainer(RawFd);
 
-impl Drop for RawFdContainer {
-    fn drop(&mut self) {
-        #[cfg(unix)]
-        {
-            use nix::unistd::close;
-            close(self.0).expect("Close failed in some RawFdContainer");
+    impl Drop for RawFdContainer {
+        fn drop(&mut self) {
+            let _ = nix::unistd::close(self.0);
         }
     }
-}
 
-impl RawFdContainer {
-    /// Create a new `RawFdContainer` for the given `RawFd`.
-    ///
-    /// The `RawFdContainer` takes ownership of the `RawFd` and closes it on drop.
-    ///
-    /// This function panics on non-unix systems.
-    pub fn new(fd: RawFd) -> RawFdContainer {
-        if cfg!(unix) {
+    impl RawFdContainer {
+        /// Create a new `RawFdContainer` for the given `RawFd`.
+        ///
+        /// The `RawFdContainer` takes ownership of the `RawFd` and closes it on drop.
+        ///
+        /// This function panics on non-unix systems.
+        pub fn new(fd: RawFd) -> Self {
             RawFdContainer(fd)
-        } else {
-            unimplemented!("RawFdContainer is only implemented on Unix-like systems");
+        }
+
+        /// Tries to clone the `RawFdContainer` creating a new FD
+        /// with `dup`. The new `RawFdContainer` will take ownership
+        /// of the `dup`ed version, whereas the original `RawFdContainer`
+        /// will keep the ownership of is FD.
+        pub fn try_clone(&self) -> nix::Result<Self> {
```

Comment:
This makes `nix` appear in the public API. I think I would like to avoid that.

I would propose something like this instead:
https://github.com/psychon/x11rb/blob/f792a01ac8e8d0c50bd5f999e287e6adb101c0c2/src/rust_connection/stream.rs#L154-L155

Reply:
Done